agent/acs/handler/payload_handler.go

```diff
@@ -274,7 +286,7 @@ func (payloadHandler *payloadRequestHandler) handleUnrecognizedTask(task *ecsacs
 	}
 
 	// Only need to stop the task; it brings down the containers too.
-	eventhandler.AddTaskEvent(api.TaskStateChange{
+	payloadHandler.taskHandler.AddTaskEvent(api.TaskStateChange{
```

Comment:
I'd prefer to leave it as is. The reasoning is that `engine.taskEvents` is being used to generate events exclusively from task engine and using that to pipe events from `payloadhandler` to the event handler seems like leaking the abstraction.